CRM/UF/Form/Field.php

```diff
@@ -551,6 +551,10 @@ public function postProcess() {
         $apiFormattedParams['location_type_id'] = $params['field_name'][2];
       }
     }
+    elseif ($params['field_name'][2] == 0) {
+      // 0 is Primary location type
+      $apiFormattedParams['location_type_id'] = NULL;
```

Comment:
@colemanw that why i have explicitly assigning  NULL value if location type is Primary ( '0' )
so that is_null always be true in this case.

is_null(CRM_Utils_Array::value('bar', $foo, ''))
i set default value '' to avoid is_null true